### PR TITLE
Add refresh weatherwidget

### LINKED ISSUE #23: Make WeatherWidget refresh its data every 5 minutes

### PR DESCRIPTION
**Pull request by jonagull**
Fix #23

### CODE REVIEW

File: src/components/widgets/WeatherWidget/useGetWeatherData.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          humidity:
+            data.properties.timeseries[0].data.instant.details
+              .relative_humidity,
+        });
+      });
+  };
+
+  const intervalRef = useRef(0);
+
+  useEffect(() => {
```
Comment (fkallevik) [inline src/components/widgets/WeatherWidget/useGetWeatherData.ts:34]: Weather data should also be fetched on page load (component mount)

**Review by fkallevik — CHANGES_REQUESTED** [review]
Get weather data on component mount

File: src/components/widgets/WeatherWidget/useGetWeatherData.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    intervalRef.current = window.setInterval(() => {
+      fetchWeatherData();
+    }, 300000);
+
+    fetchWeatherData();
+
+    return () => {
+      clearInterval(intervalRef.current);
+    };
+  }, []);
```
Comment (jonagull) [inline src/components/widgets/WeatherWidget/useGetWeatherData.ts:44]: Won't the empty dependacy array on this useEffect make the fetch happen on mount?

File: src/components/widgets/WeatherWidget/useGetWeatherData.ts (hunk 1 above)
Comment (jonagull) [reply src/components/widgets/WeatherWidget/useGetWeatherData.ts:34]: As far as I can see the dependary array for this useEffect is empty, making it fetch on mount?

Comment (fkallevik) [reply src/components/widgets/WeatherWidget/useGetWeatherData.ts:34]: Nah, it waits 5 minutes until it fetches.

File: src/components/widgets/WeatherWidget/useGetWeatherData.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  };
+
+  const intervalRef = useRef(0);
+
+  useEffect(() => {
+    intervalRef.current = window.setInterval(() => {
+      fetchWeatherData();
+    }, 300000);
+
+    fetchWeatherData();
```
Comment (jonagull) [inline src/components/widgets/WeatherWidget/useGetWeatherData.ts:39]: Tested this code again, and this call made here on like 39 fetches the data on mount, is this just a classic example of works on my machine?

Comment (fkallevik) [reply src/components/widgets/WeatherWidget/useGetWeatherData.ts:39]: Ah, sry. My bad xD

**Review by fkallevik — APPROVED** [review]

**Pull request merged**